### PR TITLE
Hipchat v2

### PR DESCRIPTION
**Pull request by frankh**
Hipchat api v1 is being shut down apparently? (I can't find a source)

We should switch to v2 anyway as it's been out for years and v1 is deprecated

### CODE REVIEW

File: cabot_alert_hipchat/models.py [hunk 1]
```diff
@@ -70,13 +71,21 @@ def _send_hipchat_alert(self, message, color='green', sender='Cabot'):
...
         api_key = env.get('HIPCHAT_API_KEY')
-        url = env.get('HIPCHAT_URL')
+        domain = env.get('HIPCHAT_DOMAIN', 'api.hipchat.com')
 
-        resp = requests.post(url + '?auth_token=' + api_key, data={
-            'room_id': room,
-            'from': sender[:15],
+        # Backwards compatibility
+        if env.get('HIPCHAT_URL'):
+            domain = urlparse(env.get('HIPCHAT_URL')).hostname
```
Comment (JeanFred) [inline cabot_alert_hipchat/models.py]: So if both `HIPCHAT_DOMAIN` and `HIPCHAT_URL` are set, the latter will take precedence?

File: cabot_alert_hipchat/tests/test_hipchat.py [hunk 2]
```diff
@@ -36,7 +34,7 @@ def test_normal_alert(self, fake_hipchat_alert):
         self.service.old_overall_status = Service.ERROR_STATUS
         self.service.save()
         self.service.alert()
-        fake_hipchat_alert.assert_called_with(u'Service Service is back to normal: http://localhost/service/1/. @test_user_hipchat_alias', color='green', sender='Cabot/Service')
+        fake_hipchat_alert.assert_called_with(u'Service Service is back to normal: http://localhost/service/1/.  @test_user_hipchat_alias', color='green', sender='Cabot/Service')
```
Comment (JeanFred) [inline cabot_alert_hipchat/tests/test_hipchat.py:37]: Any reason for the extra space?

File: setup.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 from setuptools import setup, find_packages
 
 setup(name='cabot-alert-hipchat',
-      version='1.8.3',
+      version='1.9.0',
```
Comment (JeanFred) [inline setup.py]: Shall we bump to 2.0 for the API change?

File: cabot_alert_hipchat/models.py (hunk 1 above)
Comment (frankh) [reply cabot_alert_hipchat/models.py]: Yes... I guess it should probably fail if both are set

File: cabot_alert_hipchat/tests/test_hipchat.py (hunk 2 above)
Comment (frankh) [reply cabot_alert_hipchat/tests/test_hipchat.py:37]: To get the tests to pass :)

I'm not sure when it changed but it didn't seem worth investigating

File: cabot_alert_hipchat/models.py (hunk 1 above)
Comment (frankh) [reply cabot_alert_hipchat/models.py]: Added some warnings instead

**Pull request merged**